### ISSUE TITLE
[Feature] Better organization of definition templates with multiple templates support for one trait

### ISSUE THREAD

**Issue by resouer** [oam-dev/kubevela#870]
**Is your feature request related to a problem? Please describe.**

As the title said, for now although every template/definition in KubeVela is pluggable, it's organization and installation process is still kinda coupled with KubeVela itself. That makes the contributor feels they are all "built-in" features and hesitate on making change to them, which is not the intention.

As a related issue, this makes it not straightforward that KubeVela can support multiple templates for one trait. A typical example is for Ingress, there could be various configurations for different environments, it's common to install different Ingress templates for different environments.

**Describe the solution you'd like**

One possible solution is we build relationship between environment and templates, then defer capability initialization to environment initialization. Let's say:
1. `$ vela install` will not install any capabilities by default
2. `$ vela env init demo` will accept a directory that contains environment description as well as corresponding templates, and then init the whole environment with supported workload types and traits. This step can be integrated with GitOps tool so the environment and capabilities will keep up-to-date.

In this approach, we will naturally have a layout in the environment directory to organize all templates.

A blocker is I assume we will let Trait with same name point to different templates in this case, which means per namespace environment (the current approach) won't work (unless Definition become namespaced but this seems in turn broke per cluster environment). So I personally vote for making env to cluster level.

**Describe alternatives you've considered**

**Additional context**

**Comment by wonderflow** [comment]
Good idea! The blocker doesn't exist cause we are going to change cluster scoped Traitdefinition/WorkloadDefinition into namspace scoped  #918 .

**Comment by wonderflow** [comment]
Now we are making env as a CRD #176 #160 

After that, we should implement  #918 and let the env explain which Capability the user can use.

**Comment by briankane** [comment]
This issue has been inactive for over 2 years and is scheduled for closure as part of our backlog maintenance.

  **If this issue is still relevant:**
  - Please comment below to keep it open
  - Provide updated context if the original description is outdated

  **Timeline:** If there's no activity by **Friday, October 24th**, this issue will be automatically closed.

  **Note:** Issues can always be reopened if they become relevant again. We encourage the community to reopen and reprioritize any incorrectly 
  closed issues.

**Comment by briankane** [comment]
This issue has been closed following the stale issue notice. 

If this issue is still relevant and should remain open, please:
1. Reopen the issue
2. Remove the 'stale' label
3. Provide an update on the current status or importance

Thank you for your understanding and contribution to KubeVela!
